h2o-sw-mixed-api-droplet/src/main/scala/water/droplets/H2OSWMixedAPIDroplet.scala

```diff
@@ -0,0 +1,154 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package water.droplets
+
+import hex.tree.gbm.GBMModel
+import org.apache.spark.h2o.H2OContext
+import org.apache.spark.ml.Pipeline
+import org.apache.spark.ml.feature.{HashingTF, IDF, RegexTokenizer, StopWordsRemover}
+import org.apache.spark.ml.h2o.features.ColumnPruner
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.types.{IntegerType, StringType, StructField, StructType}
+import _root_.hex.grid.GridSearch
+
+import scala.collection.JavaConverters._
+
+/**
+  * Example of Sparkling Water based application.
+  */
+object H2OSWMixedAPIDroplet {
+
+  def main(args: Array[String]) {
+
+    // Create Spark Context
+    val conf = configure("H2O-SW mixed API Droplet")
+    val spark = SparkSession.builder.config(conf).getOrCreate()
+    import spark.implicits._
+
+    // Create H2O Context
+    val hc = H2OContext.getOrCreate(spark)
+    import hc.implicits._
+
+    // Load data from file
+    val dataDF = spark.read
+      .option("delimiter", "\t")
+      .schema(StructType(Array(
+        StructField("label", StringType, nullable = false),
+        StructField("text", StringType, nullable = false),
+        StructField("weight", IntegerType, nullable = false),
+        StructField("fold", IntegerType, nullable = false))))
+      .csv("data/smsData3.txt")
+
+    val Array(trainingDF, testingDF) = dataDF.randomSplit(Array(0.9, 0.1))
+
+    // Perform pre-processing and feature engineering with Apache Spark
+    val preProcessingPipeline = sparkPreprocessingPipeline(spark).fit(trainingDF)
+
+    val preProcessedTrainingDF = preProcessingPipeline.transform(trainingDF)
+    val trainingH2OFrame = hc.asH2OFrame(preProcessedTrainingDF)
+
+    val preProcessedTestingDF = preProcessingPipeline.transform(testingDF)
+    val testingH2OFrame = hc.asH2OFrame(preProcessedTestingDF)
+
+
+    // Setup hyper-parameter search space
+    val hyperParams = Map[String, Array[AnyRef]](
+      "_ntrees" -> Array[Integer](5, 10, 15).asInstanceOf[Array[AnyRef]],
+      "_max_depth" -> Array[Integer](2, 5).asInstanceOf[Array[AnyRef]],
+      "_learn_rate" -> Array[java.lang.Double](0.01, 0.1, 0.3).asInstanceOf[Array[AnyRef]]
+    ).asJava
+
+    // Prepare training dataset to be passed to a H2O algorithm
+    val params = new GBMModel.GBMParameters
+    params._train = trainingH2OFrame.key
+    params._response_column = "label"
+    params._ignored_columns = Array("text")
+    params._weights_column = "weight"
+    params._fold_column = "fold"
+
+    trainingH2OFrame.replace(0, trainingH2OFrame.vec("label").toCategoricalVec)
+
+    // Fire off a grid search
+    val gs = GridSearch.startGridSearch(null, params, hyperParams)
+    val grid = gs.get()
+
+    // Get the best model from the grid
```

Comment:
IMHO the order returned by Grid#getModels is arbitrary, the model is thus not "best"

Reply:
WDYT about ```grid.getModels.sortBy(_.auc())(Ordering[Double].reverse).head```

Reply:
make sense to me